p2p/peer.py

```diff
@@ -735,17 +746,20 @@ def _peer_finished(self, peer: BaseService) -> None:
 
     @property
     def peers(self) -> List[BasePeer]:
-        peers = list(self.connected_nodes.values())
-        # Shuffle the list of peers so that dumb callsites are less likely to send all requests to
-        # a single peer even if they always pick the first one from the list.
-        random.shuffle(peers)
-        return peers
-
-    async def get_random_peer(self) -> BasePeer:
-        while not self.peers:
-            self.logger.debug("No connected peers, sleeping a bit")
-            await asyncio.sleep(0.5)
-        return random.choice(self.peers)
+        return list(self.connected_nodes.values())
+
+    @property
+    def highest_td_peer(self) -> BasePeer:
+        if not self.connected_nodes:
+            raise NoConnectedPeers()
+        peers_by_td = groupby(operator.attrgetter('head_td'), self.peers)
+        max_td = max(peers_by_td.keys())
+        return random.choice(peers_by_td[max_td])
+
+    def get_peers(self, min_td: int) -> List[BasePeer]:
```

Comment:
Maybe rename to `get_peers_above_height` or `get_peers_by_td`

Reply:
I'd considered that, but since the `min_td` argument is required it seemed to be clear enough already, and including that in the name would not make it any clearer. I'd prefer to leave it as is, if you're ok with that